### ISSUE TITLE
Actual refresh rate (40-50Hz) less then Monitor refresh rate (144Hz)

### ISSUE THREAD

**Issue by mgeldi** [pop-os/pop#519]
**Distribution (run `cat /etc/os-release`):**
Pop!_OS 19.04
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
Don't know what this is related to.

**Issue/Bug Description:**
I have a 144Hz Monitor. I selected the 144Hz setting both in Nvidia X Server Settings, aswell as in "Displays" within the main Settings.
In games (Rocket League), there seems to be no issue running 144Hz. It's buttery smooth. So is the cursor. The issue is that everything else on the desktop, basically the whole gnome shell itself, seems to render in sub 60Hz.
I went to [UFO test: Refresh Rate](https://www.testufo.com/refreshrate) on my browser (chromium), which showed me I only get about 48Hz refresh rate.
This issue is extremely frustrating, as I searched all over the web, not being able to find any solution to this problem. People seem to think it's a "mutter" (don't know what that is) issue. But noone actually offers a fix for this.

This is not only a problem for 144Hz Monitor users, but also for 60Hz monitor users, because even they will feel how unresponsive the desktop experience feels.

**Steps to reproduce (if you know):**
Set refresh rate -> see how windows still lag

**Expected behavior:**
Set refresh rate -> all applications (including windows, browser, etc.) run at that refresh rate

**Other Notes:**
This is a really important issue, as I feel it might scare people away from an otherwise really great distribution. I just recently discovered Pop!_OS and I really love it for it's gaming aspects. I hope someone is able to fix this and or give an easy to understand solution on how to fix it.

**Comment by mmstick** [comment]
Do you have multiple monitors? I saw this upstream: https://gitlab.gnome.org/GNOME/mutter/issues/503#note_463305

With this comment:

> I got it working by appending the following to my /etc/environment file:
>
> ```
> __GL_SYNC_DISPLAY_DEVICE=DP-2
> ```

**Comment by Rankhole** [comment]
I tried this out, instead of DP-2 I used DVI-D-0 (which is my 144Hz monitor as shown in the Nvidia Settings). I started Nvidia Settings, but nothing really seemed to have changed.

**Comment by mmstick** [comment]
Did you reboot after setting that variable?

**Comment by Rankhole** [comment]
I did now, nothing changed yet.
I should mention that I'm just using Xorg. The post mentions this only works on Wayland? So Wayland would fix my issue?
If not...I really want to keep using Pop!_OS but it's so unresponsive. Do you think changing the default desktop environment might help in any way?

**Comment by mmstick** [comment]
The comment I linked from that issue claims to have solved it for multi-monitor Xorg + NVIDIA configurations using that environment variable. I don't have a 144 Hz display on hand to test with. There's definitely an issue with Mutter and/or Xorg here.

**Comment by Rankhole** [comment]
Do I need to do anything else other than literally appending that to this environment variable?
Also, got any other suggestions on what I might try to fix this issue?

**Comment by mmstick** [comment]
Not that I know of. You may want to report [here](https://bugs.launchpad.net/ubuntu/+source/nvidia-graphics-drivers-418/+bug/1820832) that this issue affects the mutter 3.32.0 release in disco, too.

**Comment by davidbailey00** [comment]
I'm seeing an option in GNOME Settings to set my internal monitor to 120 Hz, even though it's only a 60 Hz monitor. Running `xrandr` shows the correct refresh rates. Not sure if this is the same issue.

Using an Optimus laptop (Intel UHD 630 + RTX 2080 Max-Q) and only noticed this as of the upgrade to 19.04.

**Comment by grandprixgp** [comment]
New install here, can confirm this problem still exists. I think it's mutter running at 60Hz, it doesn't bother me that much in applications but it's annoying to see that my mouse is obviously visually running at 60Hz when my monitor is at 155Hz.

**Comment by JustChristianH** [comment]
same situation over here. 144hz single display. mouse, scrolling or dragging windows etc. runs below or at 60hz. even reduced resolution to hd and hz to 120. Didn't worked. Games however make use of it.

**Comment by grandprixgp** [comment]
> 
> 
> same situation over here. 144hz single display. mouse, scrolling or dragging windows etc. runs below or at 60hz. even reduced resolution to hd and hz to 120. Didn't worked. Games however make use of it.
> 
> i agree with the OP. It"s getting more important these days where a big fraction might have 90-120 hz smartphones soon offering smooth UIs on smaller screens

I "fixed" this by moving to Wayland, which also has its issues (https://github.com/microsoft/vscode/issues/84754) but overall I found it to be a much better experience. Maybe you'd like to do the same.

**Comment by Zindswini** [comment]
Can confirm this is still an issue.
Pop 22.04, Nvidia 1070ti, 1440p 144hz primary and 1080p 60hz secondary.

Xrandr and monitor osd show 144hz active, but the difference on the desktop before and after applying the fix is huge.

This is one of those little things that is probably a super easy fix to the defaults for someone with the right skills (probably in the Nvidia proprietary :-1:  ), but dampens the experience for everyone else, especially when all of the "user friendly" tools (nvidia-settings, gnome settings, xrandr) lie to you and say everything is working as expected.

I kinda just assumed gnome wasn't as smooth as windows, but it turns out the issue was two missing config lines all along. I feel like this is worthy of a settings app tweak, or at the very least an addition to a wiki page warning people.

I really agree with the last paragraph of the original issue. If this was something someone from LTT came across, it would be the type of thing worthy of a 5 minute rant about "why the linux desktop isnt ready for gamers", when it really is sooo close. I was able to come over to Pop and not look back, mainly because of the viability of gaming. If I found this issue in the first couple weeks of switching, I probably would have jumped ship back to Windows.

**Comment by mino9421** [comment]
hardware has zero problem went from amazing performance to garbage unplayable bellow 50 fps. I cant even code properly on pop os nor ubuntu. There is something going on and nobody knows the solution yet. This is one of the reason im done with ubuntu and all its child distros for now until they figure their stuff.
I tried multiple re installation of the OS and it's still bad and does not perform.
The behaviour is similar to a machine that has no graphic drivers installed. screw gaming, but I cant even code without getting angry. I am having hard time navigating through lines of code and tabbing windows... It works, but it has miliseconds delays that are annoying and sometimes those miliseconds can be seconds.
This has been open since 2019 and I saw other similar issue unresolved.
In conclusion, good bye Ubuntu and all of its child distros until we meet again

**Comment by leviport** [comment]
Some of these sound like they could be this bug: https://github.com/pop-os/nvidia-graphics-drivers/issues/61

If you happen to be using a switchable graphics Nvidia laptop with an external display, try switching from Hybrid graphics mode to Nvidia graphics mode and see if that improves the situation. My Oryx Pro handles high refresh rate external displays wonderfully as long as it's in Nvidia mode.

**Comment by mino9421** [comment]
yes, I am using an MSI GL63 and it has amazing performance on windows, but the worst performance on ubuntu and its child distros.
I will try to switch the graphic mode and hope something works, but I know it wont work cuz ive done this many times and given up on it

**Comment by leviport** [comment]
>I will try to switch the graphic mode and hope something works, but I know it wont work cuz ive done this many times and given up on it

I mean, you don't _have_ to try it if you don't _want_ to try it :stuck_out_tongue: 
I was mostly sharing a hypothesis in case it helps anyone in the thread or anyone who finds this thread later. But I'd definitely be curious whether it improves the situation with your laptop, if you're up for it.

**Comment by mino9421** [comment]
I just gave it a shot, its better now. However, vscode working smooth, but intelliJ text navigation is sluggish. Its weird, I am exhausted of debugging the system after every few updates. I will probably go back to windows when I get the time to recover my data

**Comment by leviport** [comment]
> I will probably go back to windows when I get the time to recover my data

It sounded like you had already made up your mind to do so anyway, lol. Well thanks for confirming my theory at least. I appreciate the extra testing on a bug that has already been frustrating for you. Hopefully this info can help others.

Since this is likely a duplicate of https://github.com/pop-os/nvidia-graphics-drivers/issues/61, I'm going to close it. If I'm mistaken, we can always reopen it.
